src/main/scala/scorex/core/validation/ValidationSettings.scala

```diff
@@ -0,0 +1,33 @@
+package scorex.core.validation
+
+import scorex.core.validation.ValidationResult.Invalid
+
+/**
+  * Configuration of validation.
+  *
+  * Specifies the strategy to by used (fail-fast or error-accumulative) set of
+  * activated validation rules and their error messages
+  */
+abstract class ValidationSettings {
+  val isFailFast: Boolean
+
+  def getError(id: Short, e: Throwable): Invalid = getError(id, e.getMessage)
+
+  def getError(id: Short, details: String): Invalid
+
+  def getError(id: Short): Invalid = getError(id, "")
+
+  def isActive(id: Short): Boolean
+}
+
+class MapValidationSettings(override val isFailFast: Boolean,
```

Comment:
"MapValidationSettings" (where "Map" is a verb, which is confusing)=> "ValidationSettingsMap"

Reply:
Write comments